### ISSUE TITLE
cancelOrder seems not to work

### ISSUE THREAD

**Issue by jennazenk** [melonproject/protocol#343]
CancelOrder function from 0.5.2-alpha.1 doesn't work and estimate gas fails directly. 

https://github.com/melonproject/protocol/blob/v0.5.2-alpha.1/src/Fund.sol#L679

Assuming a fund made one order. If you call Fund.orders you get: 

```
Fund.orders = [ 
[ 6530, 
 0, 
0,
'0x2a20ff70596e431ab26C2365acab1b988DA8eCCF',
'0x1a825E9bF3BdC8ef8B975F97c78b5208a947d0EC',
1000000000000000000,
5000000000000000000,
1514969667,
0] 
]
````

I try to cancel that order by calling fund.cancelOrder(0) but it seems like something is off. Can't find the tests for that so I'm not sure if I'm calling it correctly. I also tried using the exchangeId as argument to cancelOrder.

**Comment by GauthamN** [comment]
Added a test in the latest development version. https://github.com/melonproject/protocol/blob/1659018d84ef108c3feb541984982d82a525314a/tests/integration/fundTrading.js#L638 The changes i noticed are only the use of delegatecall (which shouldn't matter) and multiple exchange adapter

**Comment by jennazenk** [comment]
Awesome thanks @GauthamN 

What is the 0 as first argument ?

https://github.com/melonproject/protocol/blob/1659018d84ef108c3feb541984982d82a525314a/tests/integration/fundTrading.js#L659

**Comment by travs** [comment]
@jennazenk 

> What is the 0 as first argument ?

Hey Jenna, that is the exchange number.
We now take into account that there may be multiple exchanges used by one fund, so we need to say which exchange to cancel the order on

**Comment by jennazenk** [comment]
Ok I see thanks @travs 

But on 0.5.2-alpha.1, was that already the case?

**Comment by travs** [comment]
@jennazenk 
Sorry forgot to mention that: no, it wasn't the case back then.
Also the test above seems to fail for me locally, so we are going to work on fixing it

**Comment by jennazenk** [comment]
https://github.com/melonproject/protocol/blob/develop/src/Fund.sol#L648

Shouldn't it be the following instead? 
````
pre_cond(isOwner() || !isShutDown)
```

**Comment by travs** [comment]
Hmm, I think the pre_cond might be okay.
The idea is that either the owner (manager) can cancel orders at any time (first condition passes), or anybody can cancel orders if the fund is shut down (second condition passes).
If we changed it to that (`... || !isShutDown`), then anybody could cancel orders while the fund is open.

We want anyone to be able to cancel orders when fund is shutdown to return funds in the case where there are outstanding orders, and the manager does not close the orders himself.
Will add a small comment to describe this since it's not clear when you first look at it.

**Comment by jennazenk** [comment]
right makes sense now! Lmk if you find the pb with cancelOrder

**Comment by retotrinkler** [comment]
ok this can be closed as soon as tests pass on latest version

**Comment by GauthamN** [comment]
Fixed in b52ab8ef8a72547dff5e0edf48ea23da270c9fac

**Comment by jennazenk** [comment]
Doesnt work yet in melon.js
